WORKSPACE.bazel

```diff
@@ -147,3 +153,15 @@ http_archive(
 load("@com_github_johnynek_bazel_jar_jar//:jar_jar.bzl", "jar_jar_repositories")
 
 jar_jar_repositories()
+
+# Declare protobuf-java from maven central, so the unshaded maven jar can recognize external dependencies
```

Comment:
Why do we care about maven being able to find protobuf, but we don't care about guava, auto:value, etc?

Reply:
We do care about those - guava and autovalue are indeed picked up by the pom automatically, as we would like them to be. Either run the build and examine the generated pom, or check out the deployed one I mentioned at https://repo1.maven.org/maven2/com/vertispan/javascript/closure-compiler-unshaded/v20220502-1/closure-compiler-unshaded-v20220502-1.pom

The difference is where closure gets those dependencies from in the bazel build - the current WORKSPACE.bazel calls `@google_bazel_common//:workspace_defs.bzl`'s `google_common_workspace_rules` (see https://github.com/google/bazel-common/blob/82a7dd0f/workspace_defs.bzl#L59-L191 for the version of the function currently in use), except here:

```
def google_common_workspace_rules():
    """Defines WORKSPACE rules for Google open-source libraries.
    Call this once at the top of your WORKSPACE file to load all of the repositories. Note that you
    should not refer to these repositories directly and instead prefer to use the targets defined in
    //third_party.
    """
    #...
    maven_import(
        group_id = "com.google.guava",
        artifact_id = "guava",
        version = "31.0.1-jre",
        licenses = ["notice"],
        sha256 = "d5be94d65e87bd219fb3193ad1517baa55a3b88fc91d21cf735826ab5af087b9",
    )
    #...
    maven_import(
        group_id = "com.google.auto.value",
        artifact_id = "auto-value",
        version = "1.6",
        licenses = ["notice"],
        sha256 = "fd811b92bb59ae8a4cf7eb9dedd208300f4ea2b6275d726e4df52d8334aaae9d",
    )
```
By virtue of using the `maven_import` defined earlier in that file (see line 55), a tag is set with the appropriate maven coords:
```
        tags = ["maven_coordinates=" + coordinates],
```

With that tag set, the `java_export` rule from `@rules_jvm_external` will correctly detect that these dependencies originally come from maven and emit dependency tags.

On the other hand, the same file also defines a rule for protobufs, which closure-compiler uses - actually it defines two (still a lie, it defines _three_, but two of the rules point to the same actual dependency), but only one is wired up with a maven coords tag:
https://github.com/google/bazel-common/blob/82a7dd0f/workspace_defs.bzl#L354-L368
```
    maven_import(
        group_id = "com.google.protobuf",
        artifact_id = "protobuf-java",
        version = "3.7.0",
        licenses = ["notice"],
        sha256 = "dc7f93e3a3dc2c11be5ba9672af3e26410f0a3289312dbf2260d4d8a0c711a51",
    )

    for protobuf_repo in ("com_google_protobuf", "com_google_protobuf_java"):
        http_archive(
            name = protobuf_repo,
            sha256 = "6b6bf5cd8d0cca442745c4c3c9f527c83ad6ef35a405f64db5215889ac779b42",
            strip_prefix = "protobuf-3.19.3",
            urls = ["https://github.com/protocolbuffers/protobuf/archive/v3.19.3.zip"],
        )

```
Since closure can't actually get by with such an old build of protobuf as 3.7.0, BUILD.bazel pointed to `@com_google_protobuf` (and uses "bazelbuild/rules_java" as `@protobuf_java_rules` to access the `java_proto_library()` def, plus "bazelbuild/rules_proto" as `@protobuf_proto_rules` for `proto_library` and co). 

(If you recall, this was the third "I'm really not sure about this, and there may well be a better way to convince bazel to do this properly" that I put in the issue, and I still suspect this to be a bug in bazel_common.)

If this is a bug in bazel_common, it has been there for a while, but it seems deliberate that these are 12 minor versions apart, and that no maven coords are attached (my understanding of bazel conventions is that there should be an alias() or something pointing at `protobuf_repo + "//:protobuf_java` (which points at https://github.com/protocolbuffers/protobuf/blob/v3.19.3/BUILD#L829-L833 fwiw), and that this generated alias would in turn have the maven_coordinates tag.

Apologies if this is too much detail for the answer, I thought this was clear in the issue, so wanted to be unambiguous.